documentation/reference/relations/db_and_db_admin.md

```diff
@@ -0,0 +1,103 @@
+|                  |            host | pgbouncer-k8s-0.pgbouncer-k8s-op…      |               |
+|                  |          master | host=pgbouncer-k8s-0.pgbouncer-k… dbname=waltz port=6432 user=relation_3  password=BjWDKjvZyClvTl4d5VDOK3mH fallback_application_name=finos-waltz  |               |
+|                  |        password | BjWDKjvZyClvTl4d5VDOK3mH               |               |
+|                  |            port | 6432                                   |               |
+|                  |        standbys | host=pgbouncer-k8s-0.pgbouncer-k… dbname=waltz port=6432 user=relation_3  password=BjWDKjvZyClvTl4d5VDOK3mH fallback_application_name=finos-waltz  |               |
+|                  |           state | master                                 |               |
+|                  |            user | relation_3                             |               |
+|                  |         version | 12.11                                  |               |
+| unit data        | allowed-subnets | 10.152.183.122/32                      |               |
+|                  |   allowed-units | pgbouncer-k8s/0                        |               |
+|                  |        database | waltz                                  | waltz         |
+|                  |            host | pgbouncer-k8s-0.pgbouncer-k8s-op…      |               |
+|                  |          master | host=pgbouncer-k8s-0.pgbouncer-k… dbname=waltz port=6432 user=relation_3 password=BjWDKjvZyClvTl4d5VDOK3mH fallback_application_name=finos-waltz  |               |
+|                  |        password | BjWDKjvZyClvTl4d5VDOK3mH               |               |
+|                  |            port | 6432                                   |               |
+|                  |        standbys | host=pgbouncer-k8s-0.pgbouncer-k…  dbname=waltz port=6432 user=relation_3  password=BjWDKjvZyClvTl4d5VDOK3mH fallback_application_name=finos-waltz  |               |
+|                  |           state | master                                 |               |
+|                  |            user | relation_3                             |               |
+|                  |         version | 12.11                                  |               |
+
+## Hook Handler Flowcharts
+
+These flowcharts detail the control flow of the hooks in this program. Unless otherwise stated, **a hook deferral is always followed by a return**.
+
+### db And db-admin Relation Joined Hook
+
+```mermaid
+flowchart TD
+  hook_fired([db-relation-joined Hook]) --> is_backend_ready{Is backend\ndatabase ready?}
+  is_backend_ready -- no --> defer>defer]
+  is_backend_ready -- yes --> is_cfg{Is pgbouncer\nconfig available?}
+  is_cfg -- no --> defer2>defer]
+  is_cfg -- yes --> extension_requested{Has the remote\napplication requested\nextensions?}
+  extension_requested -- yes --> defer3>defer\nThis charm\ncurrently doesn't\nsupport extensions]
```

Comment:
According to https://github.com/canonical/pgbouncer-k8s-operator/blob/main/src/relations/db.py#L172 it just returns (without a defer call).

Maybe add that it sets a BlockedStatus?